### PR TITLE
2.6 implement get by id member feature

### PR DESCRIPTION
**Pull request by bkmz12353**
1. Created **"GetMember(team_id, member_id)"** feature for logged in account in **ManageTeamMembersController** and **ManageTeamMembersService**
   a. Get team member by id can only team member/owner of team_id

2. Covered by tests

### CODE REVIEW

File: src/Teams/Services/ManageTeamsMembersService.cs [hunk 1]
```diff
@@ -1,8 +1,39 @@
...
+using System.Linq;
+using System.Threading.Tasks;
+
 namespace Teams.Services
 {
     public class ManageTeamsMembersService : IManageTeamsMembersService
     {
+        private readonly ICurrentUser _currentUser;
+
+        private IRepository<Team, int> _teamRepository;
```
Comment (alehkahnovich) [inline src/Teams/Services/ManageTeamsMembersService.cs:15]: readonly

File: src/Teams/Services/ManageTeamsMembersService.cs [hunk 2]
```diff
@@ -1,8 +1,39 @@
...
+            _currentUser = currentUser;
+
+            _teamRepository = teamRepository;
+
+            _teamMemberRepository = teamMemberRepository;
+        }
 
+        public  TeamMember GetMember(int team_id, string member_id)
+        {
+           return _teamMemberRepository.GetAll().Where(x => x.MemberId == member_id && x.TeamId == team_id && OwnerOrMemberOfTeam(team_id)).FirstOrDefault();
```
Comment (alehkahnovich) [inline src/Teams/Services/ManageTeamsMembersService.cs:30]: Is this is a working approach? Did you test this?

**Review by alehkahnovich — CHANGES_REQUESTED** [review]

File: Teams.Tests/MockQueryable/MockQueryable.Core/MockQueryable.Core.csproj [hunk 3]
```diff
@@ -0,0 +1,40 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (SergeyMokin) [inline Teams.Tests/MockQueryable/MockQueryable.Core/MockQueryable.Core.csproj:1]: why you added this project?

File: Teams.Tests/MockQueryable/MockQueryable.EntityFrameworkCore/MockQueryable.EntityFrameworkCore.csproj [hunk 4]
```diff
@@ -0,0 +1,46 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (SergeyMokin) [inline Teams.Tests/MockQueryable/MockQueryable.EntityFrameworkCore/MockQueryable.EntityFrameworkCore.csproj:1]: also this

File: Teams.Tests/MockQueryable/MockQueryable.Moq/MockQueryable.Moq.csproj [hunk 5]
```diff
@@ -0,0 +1,46 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (SergeyMokin) [inline Teams.Tests/MockQueryable/MockQueryable.Moq/MockQueryable.Moq.csproj:1]: this too

File: Teams.Tests/MockQueryableLibs/MoqExtensions.cs [hunk 6]
```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Microsoft.EntityFrameworkCore;
+using MockQueryable.EntityFrameworkCore;
+using Moq;
+
+namespace MockQueryable.Moq
```
Comment (SergeyMokin) [inline Teams.Tests/MockQueryableLibs/MoqExtensions.cs:9]: why you need this?
need just to install package, that is it

File: Teams.Tests/MockQueryableLibs/TestAsyncEnumerator.cs [hunk 7]
```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace MockQueryable.EntityFrameworkCore
```
Comment (SergeyMokin) [inline Teams.Tests/MockQueryableLibs/TestAsyncEnumerator.cs:5]: Same, remove it

File: Teams.Tests/MockQueryableLibs/TestExpressionVisitor.cs [hunk 8]
```diff
@@ -0,0 +1,8 @@
+﻿using System.Linq.Expressions;
+
+namespace MockQueryable.Core
+{
+	public class TestExpressionVisitor : ExpressionVisitor
```
Comment (SergeyMokin) [inline Teams.Tests/MockQueryableLibs/TestExpressionVisitor.cs:5]: this too

File: Teams.Tests/MockQueryableLibs/TestQueryProvider.cs [hunk 9]
```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+
+namespace MockQueryable.Core
```
Comment (SergeyMokin) [inline Teams.Tests/MockQueryableLibs/TestQueryProvider.cs:7]: and this

File: Teams.Tests/MockQueryableLibs/TestQueryProviderEfCore.cs [hunk 10]
```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore.Query.Internal;
+using MockQueryable.Core;
+
+namespace MockQueryable.EntityFrameworkCore
```
Comment (SergeyMokin) [inline Teams.Tests/MockQueryableLibs/TestQueryProviderEfCore.cs:10]: this too

File: Teams.Tests/Teams.Tests.csproj [hunk 11]
```diff
@@ -7,6 +7,7 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="3.1.8" />
```
Comment (SergeyMokin) [inline Teams.Tests/Teams.Tests.csproj:10]: here should be reference to MockQueryable.Core and MockQueryable.Moq
First install core

File: src/Teams/Controllers/ManageTeamMembersController.cs [hunk 12]
```diff
@@ -1,15 +1,40 @@
...
+        }
         public IActionResult Index()
         {
             return View();
         }
 
+        [Authorize, NonAction]
+        public async Task<TeamMember> GetMemberAsync(int team_id, string member_id)
+        {
+            if (!_accessCheckService.OwnerOrMemberAsync(team_id).Result)
```
Comment (SergeyMokin) [inline src/Teams/Controllers/ManageTeamMembersController.cs:31]: why Result?

Comment (SergeyMokin) [reply src/Teams/Controllers/ManageTeamMembersController.cs:31]: please read about async/await usage

File: src/Teams/Services/IAccessCheckService.cs [hunk 13]
```diff
@@ -0,0 +1,9 @@
+﻿using System.Threading.Tasks;
+
+namespace Teams.Services
+{
+    public interface IAccessCheckService
+    {
+        public  Task<bool> OwnerOrMemberAsync(int team_id);
```
Comment (SergeyMokin) [inline src/Teams/Services/IAccessCheckService.cs:7]: public?
compilation error, isn't it?
will be public by default

File: src/Teams/Services/IManageTeamsMembersService.cs [hunk 14]
```diff
@@ -1,7 +1,10 @@
-﻿namespace Teams.Services
+﻿using System.Threading.Tasks;
+using Teams.Models;
+
+namespace Teams.Services
 {
     public interface IManageTeamsMembersService
     {
-
+        public Task<TeamMember> GetMemberAsync(int team_id, string member_id);
```
Comment (SergeyMokin) [inline src/Teams/Services/IManageTeamsMembersService.cs:8]: same here

File: src/Teams/Controllers/ManageTeamMembersController.cs [hunk 15]
```diff
@@ -1,15 +1,40 @@
...
+        }
         public IActionResult Index()
         {
             return View();
         }
 
+        [Authorize, NonAction]
+        public async Task<TeamMember> GetMemberAsync(int team_id, string member_id)
+        {
+            if (await _accessCheckService.OwnerOrMemberAsync(team_id))
```
Comment (SergeyMokin) [inline src/Teams/Controllers/ManageTeamMembersController.cs:31]: please move this logic to service

File: src/Teams/Services/ManageTeamsMembersService.cs [hunk 16]
```diff
@@ -1,8 +1,29 @@
...
+        private readonly IRepository<TeamMember, int> _teamMemberRepository;
 
+        public ManageTeamsMembersService(IRepository<TeamMember, int> teamMemberRepository)
+        {
+            _teamMemberRepository = teamMemberRepository;
+        }
 
+        public async Task<TeamMember> GetMemberAsync(int team_id, string member_id)
+        {
+             return await _teamMemberRepository.GetAll()
```
Comment (SergeyMokin) [inline src/Teams/Services/ManageTeamsMembersService.cs:24]: lets see, you created new service which by logic make sense, but you can use here team repository and include team members, get a team with team members and compare result with owner and list of teammembers in team in one go

**Review by SergeyMokin — APPROVED** [review]

**Review by alehkahnovich — APPROVED** [review]

**Pull request merged**